### ISSUE TITLE
How to speed up prediction?

### ISSUE THREAD

**Issue by JackSnowWolf** [MaybeShewill-CV/CRNN_Tensorflow#185]
Hi!

When I repeat your work on my dataset, I find that prediction is extremely **slow**. I it is OK during training without decoding. But the prediction is somehow too **slow**. So do you have any suggestions to speed up inference without hardware help? Or how did you predict labels in real time? I run this model on GPU: Tesla with CUDA 9.0.

PS: I notice that the origin paper mentioned that they used BK-tree to help to speed up. In you implementation, you used CTC method and loss provided by TensorFlow. Do you have any ideas about how TensorFlow implement those functions like "ctc_beam_search_decoder"?

Thank you very much! Looking forward to your reply.

**Comment by TianboChen** [comment]
set beam_width=1 in tf.nn.ctc_beam_search_decoder

**Comment by JackSnowWolf** [comment]
> set beam_width=1 in tf.nn.ctc_beam_search_decoder
Thank You~
But I already tried this before. Do you have any idea about speed up more faster?

**Comment by TianboChen** [comment]
@JackSnowWolf What is your inference speed?

**Comment by JackSnowWolf** [comment]
@TianboChen 
i didn't calculate a accurate inference speed. The rough speed is about several minutes per batch. Or within one minute, I guess.
I may revise some implementation to get faster speed in the future. But, do you have any idea about in a high level structure?  like, in the algorithm level?
Thank you for your reply!

**Comment by MaybeShewill-CV** [comment]
@JackSnowWolf Beem search is the most time cost procedure. You may set small beam width:)
